Pi-hole Web v6.3

PromoFaux opened the pull request:
<!-- Release notes generated using configuration in .github/release.yml at development -->

## What's Changed
* Clarify client description priorities on the groups/client interface by @DL6ER in https://github.com/pi-hole/web/pull/3521
* Remove superfluous settings-level-expert class by @yubiuser in https://github.com/pi-hole/web/pull/3503
* Remove link to rate limit documentation which does not add further etails by @yubiuser in https://github.com/pi-hole/web/pull/3526
* Add some color to gravity output by @yubiuser in https://github.com/pi-hole/web/pull/3530
* settings-system : Avoid division by zero in DNS metrics by @rrobgill in https://github.com/pi-hole/web/pull/3533
* Make the "Add to denied/allowed domains" buttons red and green by @darkexplosiveqwx in https://github.com/pi-hole/web/pull/3551
* Remove query type from domainNeeded option by @yubiuser in https://github.com/pi-hole/web/pull/3557
* Handle pie chart item style conditionally by @yubiuser in https://github.com/pi-hole/web/pull/3532
* Use label-primary for number of enabled list icon on sidebar by @yubiuser in https://github.com/pi-hole/web/pull/3563
* Use innerHTML instead of append to fix gravity color output by @yubiuser in https://github.com/pi-hole/web/pull/3566
* Trim whitespaces before adding custom DNS record by @yubiuser in https://github.com/pi-hole/web/pull/3569
* Bind address to new element instead of raw HTML construction by @DL6ER in https://github.com/pi-hole/web/pull/3567
* Use /flush/network instead of /flush/arp by @yubiuser in https://github.com/pi-hole/web/pull/3553
* Prevents getting stuck loading unlimited query log results by @ablankenship10 in https://github.com/pi-hole/web/pull/3592
* Fix 2FA icon styles #3431 by @idotj in https://github.com/pi-hole/web/pull/3585
* Fix header.lp closing head tag formatting by @h3xcat in https://github.com/pi-hole/web/pull/3607
* Fix the link on the upstream chart legend by @rdwebdesign in https://github.com/pi-hole/web/pull/3606
* Rename config key dns.domain > dns.domain.name by @yubiuser in https://github.com/pi-hole/web/pull/3611
* Improve line graph tooltip by @yubiuser in https://github.com/pi-hole/web/pull/3601
* Make DNSSEC icon conditional in Queries Log (redo of https://github.com/pi-hole/web/pull/3399/) by @yubiuser in https://github.com/pi-hole/web/pull/3535
* Query Log: The underscore is special by @DL6ER in https://github.com/pi-hole/web/pull/3578
* Fix addList function to include type in API request URL by @PromoFaux in https://github.com/pi-hole/web/pull/3620
* Pin github action to SHA by @yubiuser in https://github.com/pi-hole/web/pull/3593
* Adjust domain count according to pi-hole/FTL#2177 by @yubiuser in https://github.com/pi-hole/web/pull/3619
* Also pin editorconfigchecker by sha by @yubiuser in https://github.com/pi-hole/web/pull/3621
* Update FTL %cpu and %mem everytime total CPU stats are updated by @yubiuser in https://github.com/pi-hole/web/pull/3622
* Clarify uptime in container matching the host uptime with a tooltip by @yubiuser in https://github.com/pi-hole/web/pull/3624
* Adjust text BitWarden to Bitwarden by @jprusik in https://github.com/pi-hole/web/pull/3629

## New Contributors
* @darkexplosiveqwx made their first contribution in https://github.com/pi-hole/web/pull/3551
* @ablankenship10 made their first contribution in https://github.com/pi-hole/web/pull/3592
* @idotj made their first contribution in https://github.com/pi-hole/web/pull/3585
* @h3xcat made their first contribution in https://github.com/pi-hole/web/pull/3607
* @jprusik made their first contribution in https://github.com/pi-hole/web/pull/3629

**Full Changelog**: https://github.com/pi-hole/web/compare/v6.2.1...v6.3

Commit:
charts.js: refactor to reduce complexity, improve readability and performance

Also

* drop jQuery usage
* move inline styles to CSS when possible
* switch to `a` tags for links otherwise use `p`

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
interfaces: switch to vanilla JS and refactor code to reduce complexity/improve readability

Only the bstreeview plugin is using jQuery.

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
utils: modernize hex functions

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
gravity: convert to vanilla JS and refactor code

Only the alerts are using jQuery for now

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
tailog: convert to vanilla JS and improve performance

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
tailog: validate file URL param and guard against infinite loop

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
HTML escape adlist URL before printing it in gravity stream

Co-authored-by: DL6ER <dl6er@dl6er.de>
Signed-off-by: yubiuser <github@yubiuser.dev>

Commit:
utils: remove duplicate function and convert changeTableButtonStates to vanilla JS

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

sharkboots75 committed:
Make DNSSEC icon conditional in Queries Log

Avoid emitting the DNSSEC icon in Domain column if DNSSEC is not being used/tracked

Signed-off-by: sharkboots75 <sharkboots75@gmail.com>

Commit:
Drop htmlPass util since it's only used once

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
Fix a few HTML validation issues

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
Update prettier and regenerate package-lock.json

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
Update xo to v1.1.1

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

dependabot[bot] committed:
build(deps-dev): bump postcss from 8.5.3 to 8.5.4

Bumps [postcss](https://github.com/postcss/postcss) from 8.5.3 to 8.5.4.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/8.5.3...8.5.4)

---
updated-dependencies:
- dependency-name: postcss
  dependency-version: 8.5.4
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
header_authenticated.lp: add the hostname `li` only if it's greater than zero

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
footer: use text for sysinfo-cpu

There's no HTML involved.

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
footer: use text for sysinfo-cpu (#3502)

yubiuser committed:
Remove superfluosus ettings-level-expert class

Signed-off-by: yubiuser <github@yubiuser.dev>

dependabot[bot] committed:
build(deps): bump github/codeql-action from 3.28.18 to 3.28.19

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 3.28.18 to 3.28.19.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v3.28.18...v3.28.19)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 3.28.19
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
build(deps): bump github/codeql-action from 3.28.18 to 3.28.19 (#3507)

yubiuser committed:
build(deps-dev): bump postcss from 8.5.3 to 8.5.4 (#3498)

yubiuser committed:
Sync master back into development (#3496)

dependabot[bot] committed:
build(deps): bump github/codeql-action from 3.28.19 to 3.29.0

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 3.28.19 to 3.29.0.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v3.28.19...v3.29.0)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 3.29.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump postcss from 8.5.4 to 8.5.5

Bumps [postcss](https://github.com/postcss/postcss) from 8.5.4 to 8.5.5.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/8.5.4...8.5.5)

---
updated-dependencies:
- dependency-name: postcss
  dependency-version: 8.5.5
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
settings-dns: expand custom servers box if it has custom servers

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
build(deps-dev): bump postcss from 8.5.4 to 8.5.5 (#3512)

yubiuser committed:
build(deps): bump github/codeql-action from 3.28.19 to 3.29.0 (#3510)

Commit:
header: don't hide advanced info since it's always shown

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
Lowercase doctype

It's perfectly valid and it should result in smaller compressed size when gzip is used.

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
Lowercase doctype (#3515)

yubiuser committed:
header: don't hide advanced info since it's always shown (#3513)

yubiuser committed:
Fix a few HTML validation issues (#3517)

Commit:
Update chart.js to v4.5.0 (#3516)

* Update chart.js to v4.5.0

Also, switch to the minified file

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

* Update scripts/lua/header_authenticated.lp

Co-authored-by: yubiuser <github@yubiuser.dev>
Signed-off-by: XhmikosR <xhmikosr@gmail.com>

---------

Signed-off-by: XhmikosR <xhmikosr@gmail.com>
Co-authored-by: yubiuser <github@yubiuser.dev>

Commit:
footer: remove document.hidden in checkBlocking (#3518)

This is taken care of in utils.setTimer.

Fixes status not showing when opening a new tab. Previously, it was shown after the timer fired.

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
header: move unneeded unauthenticated assets to authenticated

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

DL6ER committed:
Clarify client description priorities on the groups/client interface

Signed-off-by: DL6ER <dl6er@dl6er.de>

DL6ER committed:
Clarify client description priorities on the groups/client interface (#3521)

DL6ER committed:
header_authenticated.lp: add the hostname `li` only if it's greater t… (#3501)

yubiuser committed:
Remove superfluous settings-level-expert class (#3503)

Remove superfluosus ettings-level-expert class

Signed-off-by: yubiuser <github@yubiuser.dev>

Commit:
Abstract collapsing box to a new util

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

Commit:
groups-clients.lp: fix invalid HTML

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
groups-clients.lp: fix invalid HTML (#3522)

dependabot[bot] committed:
build(deps-dev): bump postcss from 8.5.5 to 8.5.6

Bumps [postcss](https://github.com/postcss/postcss) from 8.5.5 to 8.5.6.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/8.5.5...8.5.6)

---
updated-dependencies:
- dependency-name: postcss
  dependency-version: 8.5.6
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
build(deps-dev): bump postcss from 8.5.5 to 8.5.6 (#3524)

yubiuser committed:
Remove link to rate limit documentation which does not add further details

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Remove link to rate limit documentation which does not add further etails (#3526)

yubiuser committed:
Add some color to gravity output

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Replace ANSI escape codes with HTML tags

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Add <span> counter

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Disable eslint prettier/prettier rule for the block too keep it readable

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Use mapping

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Processs line squentially to always close the right number of <span> once we hit {COL_NC}

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Only underline clickable legend items in pie charts

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Show pointer only for clickable items

Signed-off-by: yubiuser <github@yubiuser.dev>

DL6ER committed:
Add some color to gravity output (#3530)

rrobgill committed:
settings-system : Avoid division by zero in DNS metrics

Don't attempt to calculate a percentage if there are no replies yet

Signed-off-by: Rob Gill <rrobgill@protonmail.com>

yubiuser committed:
settings-system : Avoid division by zero in DNS metrics (#3533)

yubiuser committed:
Fix tests

Signed-off-by: yubiuser <github@yubiuser.dev>

Commit:
settings-dns-records: merge DOMContentLoaded listeners

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

yubiuser committed:
header: move unneeded unauthenticated assets to authenticated (#3536)

yubiuser committed:
Update prettier and regenerate package-lock.json (#3537)

yubiuser committed:
Drop htmlPass util since it's only used once (#3539)

yubiuser committed:
settings-dns-records: merge DOMContentLoaded listeners (#3538)

Commit:
secondsTimeSpanToHMS: use modulo operator and padStart

Signed-off-by: XhmikosR <xhmikosr@gmail.com>

dependabot[bot] committed:
build(deps): bump github/codeql-action from 3.29.0 to 3.29.1

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 3.29.0 to 3.29.1.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v3.29.0...v3.29.1)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 3.29.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
build(deps): bump github/codeql-action from 3.29.0 to 3.29.1 (#3549)

darkexplosiveqwx committed:
Apply red and green to "Add to  domains" buttons

Signed-off-by: darkexplosiveqwx <101737077+darkexplosiveqwx@users.noreply.github.com>

yubiuser committed:
Make the "Add to denied/allowed domains" buttons red and green (#3551)

PromoFaux committed:
settings-dns: expand custom servers box if it has custom servers (#3514)

yubiuser committed:
Use /flush/network instead of /flush/arp

Signed-off-by: yubiuser <github@yubiuser.dev>

dependabot[bot] committed:
build(deps): bump github/codeql-action from 3.29.1 to 3.29.2

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 3.29.1 to 3.29.2.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v3.29.1...v3.29.2)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 3.29.2
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
build(deps): bump github/codeql-action from 3.29.1 to 3.29.2 (#3556)

yubiuser committed:
Remove query type from domainNeeded option

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Remove query type from domainNeeded option (#3557)

PromoFaux committed:
Update xo to v1.1.1 (#3485)

PromoFaux committed:
secondsTimeSpanToHMS: use modulo operator and padStart (#3548)

PromoFaux committed:
taillog: convert to vanilla JS and improve things (#3550)

PromoFaux committed:
utils: remove duplicate function and convert changeTableButtonStates to vanilla JS (#3547)

PromoFaux committed:
utils: modernize hex functions (#3546)

PromoFaux committed:
gravity: convert to vanilla JS and refactor code (#3545)

PromoFaux committed:
interfaces: switch to vanilla JS and refactor code to reduce complexi… (#3544)

PromoFaux committed:
Sync master back into development (#3558)

PromoFaux committed:
charts.js: refactor to reduce complexity, improve readability and per… (#3542)

PromoFaux committed:
Handle pie chart item style conditionally (#3532)

dependabot[bot] committed:
build(deps-dev): bump eslint from 9.29.0 to 9.31.0

Bumps [eslint](https://github.com/eslint/eslint) from 9.29.0 to 9.31.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v9.29.0...v9.31.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-version: 9.31.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
build(deps-dev): bump eslint from 9.29.0 to 9.31.0 (#3560)

dependabot[bot] committed:
build(deps-dev): bump globals from 16.2.0 to 16.3.0

Bumps [globals](https://github.com/sindresorhus/globals) from 16.2.0 to 16.3.0.
- [Release notes](https://github.com/sindresorhus/globals/releases)
- [Commits](https://github.com/sindresorhus/globals/compare/v16.2.0...v16.3.0)

---
updated-dependencies:
- dependency-name: globals
  dependency-version: 16.3.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
build(deps-dev): bump globals from 16.2.0 to 16.3.0 (#3561)

rdwebdesign committed:
Replace `mg.request_info.request_uri` with the variable `scriptname`

The information from `mg.request_info.request_uri` depends on the URL typed
by the user. This information was used without any sanitization, allowing
an attacker to send crafted links containing anything, including javascript
code, which could be loaded and executed in a few pages.

Replacing this value with `scriptname` variable fixes the issue, since this
variable contains the name of the file currently being executed. This
information cannot be externally manipulated and it is safe to be used on
the page.

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

yubiuser committed:
Use label-primary for number of enabled list icon on sidebar

Signed-off-by: yubiuser <github@yubiuser.dev>

DL6ER committed:
Use label-primary for number of enabled list icon on sidebar (#3563)

yubiuser committed:
Use innerHTML instead of append to fix gravity color output

Signed-off-by: yubiuser <github@yubiuser.dev>

DL6ER committed:
Use innerHTML instead of append to fix gravity color output (#3566)

DL6ER committed:
Bind address to new element instead of raw HTML construction

Signed-off-by: DL6ER <dl6er@dl6er.de>

yubiuser committed:
Trim whitespaces before adding custom DNS record

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Also trim CNAME input fields

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Trim whitespaces before adding custom DNS record (#3569)

DL6ER committed:
Bind address to new element instead of raw HTML construction (#3567)

dependabot[bot] committed:
build(deps-dev): bump xo from 1.1.1 to 1.2.1

Bumps [xo](https://github.com/xojs/xo) from 1.1.1 to 1.2.1.
- [Release notes](https://github.com/xojs/xo/releases)
- [Commits](https://github.com/xojs/xo/compare/v1.1.1...v1.2.1)

---
updated-dependencies:
- dependency-name: xo
  dependency-version: 1.2.1
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
build(deps-dev): bump xo from 1.1.1 to 1.2.1 (#3570)

yubiuser committed:
Use \u001B instead of (hidden) ESC character

Signed-off-by: yubiuser <github@yubiuser.dev>

dependabot[bot] committed:
build(deps-dev): bump eslint from 9.31.0 to 9.32.0

Bumps [eslint](https://github.com/eslint/eslint) from 9.31.0 to 9.32.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v9.31.0...v9.32.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-version: 9.32.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github/codeql-action from 3.29.2 to 3.29.4

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 3.29.2 to 3.29.4.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v3.29.2...v3.29.4)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 3.29.4
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
build(deps): bump github/codeql-action from 3.29.2 to 3.29.4 (#3574)

yubiuser committed:
build(deps-dev): bump eslint from 9.31.0 to 9.32.0 (#3572)

dependabot[bot] committed:
build(deps): bump github/codeql-action from 3.29.4 to 3.29.5

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 3.29.4 to 3.29.5.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v3.29.4...v3.29.5)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 3.29.5
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
build(deps): bump github/codeql-action from 3.29.4 to 3.29.5 (#3576)

DL6ER committed:
Hint at the special role of the underscore character

Signed-off-by: DL6ER <dl6er@dl6er.de>

dependabot[bot] committed:
build(deps-dev): bump eslint from 9.32.0 to 9.33.0

Bumps [eslint](https://github.com/eslint/eslint) from 9.32.0 to 9.33.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v9.32.0...v9.33.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-version: 9.33.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github/codeql-action from 3.29.5 to 3.29.8

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 3.29.5 to 3.29.8.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v3.29.5...v3.29.8)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 3.29.8
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
build(deps-dev): bump eslint from 9.32.0 to 9.33.0 (#3579)

yubiuser committed:
build(deps): bump github/codeql-action from 3.29.5 to 3.29.8 (#3580)

yubiuser committed:
Use /flush/network instead of /flush/arp (#3553)

dependabot[bot] committed:
build(deps-dev): bump xo from 1.2.1 to 1.2.2

Bumps [xo](https://github.com/xojs/xo) from 1.2.1 to 1.2.2.
- [Release notes](https://github.com/xojs/xo/releases)
- [Commits](https://github.com/xojs/xo/compare/v1.2.1...v1.2.2)

---
updated-dependencies:
- dependency-name: xo
  dependency-version: 1.2.2
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github/codeql-action from 3.29.8 to 3.29.9

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 3.29.8 to 3.29.9.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v3.29.8...v3.29.9)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 3.29.9
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump actions/checkout from 4.2.2 to 5.0.0

Bumps [actions/checkout](https://github.com/actions/checkout) from 4.2.2 to 5.0.0.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v4.2.2...v5.0.0)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: 5.0.0
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
build(deps-dev): bump xo from 1.2.1 to 1.2.2 (#3582)

yubiuser committed:
build(deps): bump actions/checkout from 4.2.2 to 5.0.0 (#3584)

yubiuser committed:
build(deps): bump github/codeql-action from 3.29.8 to 3.29.9 (#3583)

idotj committed:
Fix 2FA icon styles #3431

Signed-off-by: idotj <info@idotj.com>

idotj committed:
Fix typo

Signed-off-by: idotj <info@idotj.com>

idotj committed:
Remove CSS

Signed-off-by: idotj <info@idotj.com>

dependabot[bot] committed:
build(deps-dev): bump eslint from 9.33.0 to 9.34.0

Bumps [eslint](https://github.com/eslint/eslint) from 9.33.0 to 9.34.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v9.33.0...v9.34.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-version: 9.34.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github/codeql-action from 3.29.9 to 3.29.11

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 3.29.9 to 3.29.11.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v3.29.9...v3.29.11)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 3.29.11
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
build(deps-dev): bump eslint from 9.33.0 to 9.34.0 (#3589)

yubiuser committed:
build(deps): bump github/codeql-action from 3.29.9 to 3.29.11 (#3591)

ablankenship10 committed:
Prevents getting stuck loading unlimited query log results. Changed in Queries Log table only.

Signed-off-by: Aaron Blankenship <aaron@aaronblankenship.com>

PromoFaux committed:
Prevents getting stuck loading unlimited query log results (#3592)

rdwebdesign committed:
Fix 2FA icon styles #3431 (#3585)

yubiuser committed:
Pin github action to SHA

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser reviewed (commented):
Need to add the security related PRs!

yubiuser committed:
Show tooltip of bar graphs below x-axis

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Only display items in queryOverTimeChart tooltips with at least 1 query

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Fix calculation of tootlip percentage

Signed-off-by: yubiuser <github@yubiuser.dev>

rdwebdesign committed:
Remove unnecessary code

The outer `if` (line 92) already guarantees only 2 possible values.
If `isQueryTypeChart` is false, `isForwardDestinationChart` must be true.

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign committed:
Use an array to store upstream server IPs

This will avoid overwritting the IP when more than one upstream DNS server uses the same name

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign committed:
Show the upstream server IP on the title tooltip

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign committed:
Include the upstream DNS server name to the link, when needed

The link must send the "upstream" parameter in exactly the same format used
by the "suggestions" API.
The format is: "upstream=<IP>#<port> (<servername>)".
This will ensure that when a link is clicked, the correct server name will
be highlighted in the SELECT element on the queries.lp page and no other
OPTION element will be created.

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

h3xcat committed:
Fix closing head tag formatting

Signed-off-by: H3xCat <2304691+h3xcat@users.noreply.github.com>

rdwebdesign committed:
Fix closing head tag (#3607)

Fixed missing `</head>` tag: files erro403.lp, error404.lp and login.lp

rdwebdesign committed:
Add offset effect on hover to the doughnut charts

We need to add a small padding to avoid "clipping" the arc/slice.
This happens because when an arc/slice expands, it grows beyond the canvas
limits and get clipped.

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign committed:
Trigger the offset effect when the mouse is over a legend item

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

rdwebdesign committed:
Fix the link on the upstream chart legend (#3606)

yubiuser committed:
Rename config key dns.domain > dns.domain.name

Signed-off-by: yubiuser <github@yubiuser.dev>

yubiuser committed:
Rename config key dns.domain > dns.domain.name (#3611)

yubiuser committed:
Improve line graph tooltip (#3601)

DL6ER committed:
Improve wording

Co-authored-by: Adam Warner <me@adamwarner.co.uk>
Signed-off-by: Dominik <DL6ER@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump eslint from 9.34.0 to 9.37.0

Bumps [eslint](https://github.com/eslint/eslint) from 9.34.0 to 9.37.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Commits](https://github.com/eslint/eslint/compare/v9.34.0...v9.37.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-version: 9.37.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

PromoFaux committed:
Make DNSSEC icon conditional in Queries Log (redo of https://github.com/pi-hole/web/pull/3399/) (#3535)

DL6ER committed:
Query Log: The underscore is special (#3578)

yubiuser committed:
Adjust domain count according to https://github.com/pi-hole/FTL/pull/2177

Signed-off-by: yubiuser <github@yubiuser.dev>

PromoFaux committed:
Fix addList function to include type in API request URL

Signed-off-by: Adam Warner <me@adamwarner.co.uk>

PromoFaux committed:
Fix addList function to include type in API request URL (#3620)

yubiuser committed:
Pin github action to SHA (#3593)

dependabot[bot] committed:
build(deps): bump actions/setup-node from 4.4.0 to 5.0.0

Bumps [actions/setup-node](https://github.com/actions/setup-node) from 4.4.0 to 5.0.0.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/v4.4.0...v5.0.0)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-version: 5.0.0
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump actions/stale from 9.1.0 to 10.1.0

Bumps [actions/stale](https://github.com/actions/stale) from 9.1.0 to 10.1.0.
- [Release notes](https://github.com/actions/stale/releases)
- [Changelog](https://github.com/actions/stale/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/stale/compare/v9.1.0...v10.1.0)

---
updated-dependencies:
- dependency-name: actions/stale
  dependency-version: 10.1.0
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github/codeql-action from 3.29.11 to 3.30.6

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 3.29.11 to 3.30.6.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v3.29.11...v3.30.6)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 3.30.6
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
build(deps): bump github/codeql-action from 3.29.11 to 3.30.6 (#3616)

yubiuser committed:
build(deps): bump actions/stale from 9.1.0 to 10.1.0 (#3615)

yubiuser committed:
build(deps): bump actions/setup-node from 4.4.0 to 5.0.0 (#3597)

yubiuser committed:
build(deps-dev): bump eslint from 9.34.0 to 9.37.0 (#3618)

dependabot[bot] committed:
build(deps-dev): bump globals from 16.3.0 to 16.4.0

Bumps [globals](https://github.com/sindresorhus/globals) from 16.3.0 to 16.4.0.
- [Release notes](https://github.com/sindresorhus/globals/releases)
- [Commits](https://github.com/sindresorhus/globals/compare/v16.3.0...v16.4.0)

---
updated-dependencies:
- dependency-name: globals
  dependency-version: 16.4.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
build(deps-dev): bump globals from 16.3.0 to 16.4.0 (#3605)

yubiuser committed:
Also pin editorconfigchecker by sha

Signed-off-by: yubiuser <github@yubiuser.dev>

DL6ER committed:
Adjust domain count according to pi-hole/FTL#2177 (#3619)

yubiuser committed:
Also pin editorconfigchecker by sha (#3621)

yubiuser committed:
Update FTL %cpu and %mem everytime total CPU stats are updated

Signed-off-by: yubiuser <github@yubiuser.dev>

rdwebdesign committed:
Add CPU usage percentage to the Load tooltip

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

DL6ER committed:
Update FTL %cpu and %mem everytime total CPU stats are updated (#3622)

yubiuser committed:
Clarify uptime in container matching the host uptime with a tooltip

Signed-off-by: yubiuser <github@yubiuser.dev>

dependabot[bot] committed:
build(deps): bump editorconfig-checker/action-editorconfig-checker

Bumps [editorconfig-checker/action-editorconfig-checker](https://github.com/editorconfig-checker/action-editorconfig-checker) from f40bac9e7d9e7d298fbe36b83e1eff8f0de13fb8 to 1a41284d59c6fe7f1b21ddc4a2b36400a33dc1b4.
- [Release notes](https://github.com/editorconfig-checker/action-editorconfig-checker/releases)
- [Commits](https://github.com/editorconfig-checker/action-editorconfig-checker/compare/f40bac9e7d9e7d298fbe36b83e1eff8f0de13fb8...1a41284d59c6fe7f1b21ddc4a2b36400a33dc1b4)

---
updated-dependencies:
- dependency-name: editorconfig-checker/action-editorconfig-checker
  dependency-version: 1a41284d59c6fe7f1b21ddc4a2b36400a33dc1b4
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github/codeql-action from 3.30.6 to 4.30.8

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 3.30.6 to 4.30.8.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/64d10c13136e1c5bce3e5fbde8d4906eeaafc885...f443b600d91635bebf5b0d9ebc620189c0d6fba5)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 4.30.8
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump xo from 1.2.2 to 1.2.3

Bumps [xo](https://github.com/xojs/xo) from 1.2.2 to 1.2.3.
- [Release notes](https://github.com/xojs/xo/releases)
- [Commits](https://github.com/xojs/xo/compare/v1.2.2...v1.2.3)

---
updated-dependencies:
- dependency-name: xo
  dependency-version: 1.2.3
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
build(deps): bump editorconfig-checker/action-editorconfig-checker from f40bac9e7d9e7d298fbe36b83e1eff8f0de13fb8 to 1a41284d59c6fe7f1b21ddc4a2b36400a33dc1b4 (#3625)

yubiuser committed:
build(deps): bump github/codeql-action from 3.30.6 to 4.30.8 (#3626)

yubiuser committed:
build(deps-dev): bump xo from 1.2.2 to 1.2.3 (#3628)

yubiuser committed:
Clarify uptime in container matching the host uptime with a tooltip (#3624)

jprusik committed:
Adjust BitWarden to Bitwarden

Signed-off-by: Jonathan Prusik <jprusik@classynemesis.com>

yubiuser committed:
Adjust text BitWarden to Bitwarden (#3629)

yubiuser committed:
Escape all lines to also prevent XSS for non-domain entries

Signed-off-by: yubiuser <github@yubiuser.dev>

dependabot[bot] committed:
build(deps): bump actions/setup-node from 5.0.0 to 6.0.0

Bumps [actions/setup-node](https://github.com/actions/setup-node) from 5.0.0 to 6.0.0.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/a0853c24544627f65ddf259abe73b1d18a591444...2028fbc5c25fe9cf00d9f06a71cc4710d4507903)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-version: 6.0.0
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump editorconfig-checker/action-editorconfig-checker

Bumps [editorconfig-checker/action-editorconfig-checker](https://github.com/editorconfig-checker/action-editorconfig-checker) from 1a41284d59c6fe7f1b21ddc4a2b36400a33dc1b4 to 5ecdd656fe347c26f76b1b435b90e1d74fb5e787.
- [Release notes](https://github.com/editorconfig-checker/action-editorconfig-checker/releases)
- [Commits](https://github.com/editorconfig-checker/action-editorconfig-checker/compare/1a41284d59c6fe7f1b21ddc4a2b36400a33dc1b4...5ecdd656fe347c26f76b1b435b90e1d74fb5e787)

---
updated-dependencies:
- dependency-name: editorconfig-checker/action-editorconfig-checker
  dependency-version: 5ecdd656fe347c26f76b1b435b90e1d74fb5e787
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github/codeql-action from 4.30.8 to 4.30.9

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 4.30.8 to 4.30.9.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/f443b600d91635bebf5b0d9ebc620189c0d6fba5...16140ae1a102900babc80a33c44059580f687047)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 4.30.9
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump eslint from 9.37.0 to 9.38.0

Bumps [eslint](https://github.com/eslint/eslint) from 9.37.0 to 9.38.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Commits](https://github.com/eslint/eslint/compare/v9.37.0...v9.38.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-version: 9.38.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

yubiuser committed:
build(deps): bump actions/setup-node from 5.0.0 to 6.0.0 (#3632)

yubiuser committed:
build(deps): bump editorconfig-checker/action-editorconfig-checker from 1a41284d59c6fe7f1b21ddc4a2b36400a33dc1b4 to 5ecdd656fe347c26f76b1b435b90e1d74fb5e787 (#3633)

yubiuser committed:
build(deps): bump github/codeql-action from 4.30.8 to 4.30.9 (#3634)

yubiuser committed:
build(deps-dev): bump eslint from 9.37.0 to 9.38.0 (#3635)

rdwebdesign committed:
Remove `mg.request_info.request_uri` from error403.lp

This page is currently not used because we redirect to login page in case
of unauthenticated access, but I'm removing it just in case we decide to
change how we handle 403 errors in the future.

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

PromoFaux committed:
Merge commit from fork

Replace `mg.request_info.request_uri` with the variable `scriptname`

PromoFaux committed:
Merge commit from fork

HTML escape adlist URL and non-domain entried before printing it in gravity stream
